src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -45,21 +75,46 @@ function SearchRouter() {
         }
     }, SEARCH_DEBOUNCE_DELAY);
 
-    const onSearchSubmit = useCallback(() => {
+    const closeAndClearRouter = useCallback(() => {
         closeSearchRouter();
-
-        const query = SearchUtils.buildSearchQueryString(currentQuery);
-        Navigation.navigate(ROUTES.SEARCH_CENTRAL_PANE.getRoute({query}));
         clearUserQuery();
-    }, [currentQuery, closeSearchRouter]);
+    }, [closeSearchRouter]);
+
+    const onSearchSubmit = useCallback(
+        (query: SearchQueryJSON | undefined) => {
+            if (!query) {
+                return;
+            }
+            closeSearchRouter();
+            const queryString = SearchUtils.buildSearchQueryString(query);
+            Navigation.navigate(ROUTES.SEARCH_CENTRAL_PANE.getRoute({query: queryString}));
+            clearUserQuery();
+        },
+        [closeSearchRouter],
+    );
 
+<<<<<<< HEAD
+    useKeyboardShortcut(
+        CONST.KEYBOARD_SHORTCUTS.ENTER,
```

Comment:
I think we still need to handle ENTER in this case, because SelectionList(in BaseSelectionList) is catching ENTER shortcut and therefore it cannot be caught by SearchInput